src/test/scala/uk/gov/nationalarchives/LambdaTest.scala

```diff
@@ -0,0 +1,236 @@
+
+    private val asyncS3Client: S3AsyncClient = S3AsyncClient
+      .crtBuilder()
+      .endpointOverride(URI.create("http://localhost:9006"))
+      .region(Region.EU_WEST_2)
+      .credentialsProvider(creds)
+      .targetThroughputInGbps(20.0)
+      .minimumPartSizeInBytes(10 * 1024 * 1024)
+      .build()
+    override val dynamoClient: DADynamoDBClient[IO] = new DADynamoDBClient[IO](asyncDynamoClient)
+    override val s3Client: DAS3Client[IO] = DAS3Client[IO](asyncS3Client)
+  }
+
+  "handleRequest" should "return an error if the folder is not found in dynamo" in {
+    stubGetRequest(emptyDynamoGetResponse)
+    val ex = intercept[Exception] {
+      TestLambda().handleRequest(standardInput, outputStream, null)
+    }
+    ex.getMessage should equal(s"No folder found for $folderId and $batchId")
+  }
+
+  "handleRequest" should "return an error if no children are found for the folder" in {
+    stubGetRequest(dynamoGetResponse)
+    stubQueryRequest(emptyDynamoQueryResponse)
+    val ex = intercept[Exception] {
+      TestLambda().handleRequest(standardInput, outputStream, null)
+    }
+    ex.getMessage should equal(s"No children found for $folderId and $batchId")
+  }
+
+  "handleRequest" should "return an error if the dynamo entry does not have a type of 'folder'" in {
+    stubGetRequest(dynamoGetResponse.replace("Folder", "Asset"))
+    stubQueryRequest(emptyDynamoQueryResponse)
+    val ex = intercept[Exception] {
+      TestLambda().handleRequest(standardInput, outputStream, null)
+    }
+    ex.getMessage should equal(s"Object $folderId is of type Asset and not 'folder'")
+  }
+
+  "handleRequest" should "pass the correct id to dynamo getItem" in {
+    stubGetRequest(emptyDynamoGetResponse)
+    intercept[Exception] {
+      TestLambda().handleRequest(standardInput, outputStream, null)
+    }
+    val serveEvents = dynamoServer.getAllServeEvents.asScala
+    serveEvents.size should equal(1)
+    serveEvents.head.getRequest.getBodyAsString should equal(s"""{"RequestItems":{"test-table":{"Keys":[{"id":{"S":"$folderId"}}]}}}""")
+  }
+
+  "handleRequest" should "pass the correct parameters to dynamo for the query request" in {
+    stubGetRequest(dynamoGetResponse)
+    stubQueryRequest(emptyDynamoQueryResponse)
+    intercept[Exception] {
+      TestLambda().handleRequest(standardInput, outputStream, null)
+    }
+    val serveEvents = dynamoServer.getAllServeEvents.asScala
+    val queryEvent = serveEvents.head
+    val requestBody = queryEvent.getRequest.getBodyAsString
+    val expectedRequestBody =
+      """{"TableName":"test-table","IndexName":"test-gsi","KeyConditionExpression":"#A = :batchId AND #B = :parentPath",""" +
+        s""""ExpressionAttributeNames":{"#A":"batchId","#B":"parentPath"},"ExpressionAttributeValues":{":batchId":{"S":"TEST-ID"},":parentPath":{"S":"$folderParentPath/$folderId"}}}"""
+    expectedRequestBody should equal(requestBody)
+  }
+
+  "handleRequest" should "upload the opex file" in {
```

Comment:
This is only testing the path; do you think:

1. this title should be changed to "upload the opex file using the correct url" or something like that
2. it's worth adding a test to see if the correct opex is sent (either here or a brand new test)?

Reply:
I've changed the title of this test and I've added another one in which checks what is sent to S3.